cellengine/resources/gate.py

```diff
@@ -193,81 +152,345 @@ def create(
+            fcs_file (str, optional): Name of FCS file, if tailored per file.
+                An attempt will be made to find the file by name. If zero or
+                more than one file exists with the name, an error will be
+                thrown. Looking up files by name is slower than using the ID,
+                as this requires additional requests to the server. If
+                specified, do not specify ``fcs_file_id``.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
+
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+        kwargs = get_args_as_kwargs(cls, locals())
+        params = {
+            k: kwargs.pop(k) for k in ["create_population", "parent_population_id"]
+        }
+
+        gate = cls._format(**kwargs)
+        return ce.APIClient().create(Gate(id=None, **gate), **params)  # type: ignore
 
     @classmethod
-    @doc_inherit(format_polygon_gate)
+    @exception_handler
+    def _format(cls, **kwargs):
+        """Get relevent kwargs and shape into a gate model"""
+
+        kwargs = remove_keys_with_none_values(kwargs)
+
+        x1 = kwargs.get("x1") or kwargs.get("model", {}).get("rectangle", {}).get("x1")
+        x2 = kwargs.get("x2") or kwargs.get("model", {}).get("rectangle", {}).get("x2")
+        y1 = kwargs.get("y1") or kwargs.get("model", {}).get("rectangle", {}).get("y1")
+        y2 = kwargs.get("y2") or kwargs.get("model", {}).get("rectangle", {}).get("y2")
+        label = kwargs.get("label") or kwargs.get("model", {}).get(
+            "label",
+            [
+                mean([x1, x2]),
+                mean([y1, y2]),
+            ],
+        )
+
+        model = {
+            "locked": kwargs.get("model", {}).get("locked", False),
+            "label": label,
+            "rectangle": {"x1": x1, "x2": x2, "y1": y1, "y2": y2},
+        }
+
+        return {
+            "experiment_id": kwargs["experiment_id"],
+            "fcs_file_id": parse_fcs_file_args(
+                kwargs.get("experiment_id"),
+                kwargs.get("tailored_per_file", False),
+                kwargs.get("fcs_file_id"),
+                kwargs.get("fcs_file"),
+            ),
+            "gid": kwargs.get("gid", generate_id()),
+            "model": model,
+            "name": kwargs.get("name"),
+            "tailored_per_file": kwargs.get("tailored_per_file", False),
+            "type": "RectangleGate",
+            "x_channel": kwargs.get("x_channel"),
+            "y_channel": kwargs.get("y_channel"),
+        }
+
+
+class PolygonGate(Gate):
```

Comment:
(only reviewed Rectangle, didn't review any of the code in this file from here-down)